### PR TITLE
Add support for access token using "secret text" credentials of Jenkins Credentials Plugin.

### PR DESCRIPTION
**Pull request by liron-shalom**

### CODE REVIEW

File: src/main/java/org/jfrog/hudson/pipeline/common/docker/DockerImage.java [hunk 1]
```diff
@@ -134,12 +134,12 @@ public Module generateBuildInfoModule(Run build, TaskListener listener, Artifact
         try {
             dependenciesClient = server.createArtifactoryDependenciesClient(
                     preferredResolver.provideUsername(build.getParent()), preferredResolver.providePassword(build.getParent()),
-                    server.createProxyConfiguration(Jenkins.getInstance().proxy), listener);
+                    preferredResolver.provideAccessToken(build.getParent()), server.createProxyConfiguration(Jenkins.getInstance().proxy), listener);
```
Comment (eyalbe4) [inline src/main/java/org/jfrog/hudson/pipeline/common/docker/DockerImage.java]: What is the reason for this change?

File: src/main/java/org/jfrog/hudson/pipeline/common/docker/DockerImage.java [hunk 2]
```diff
@@ -134,12 +134,12 @@ public Module generateBuildInfoModule(Run build, TaskListener listener, Artifact
...
 
             CredentialsConfig preferredDeployer = CredentialManager.getPreferredDeployer(config, server);
             propertyChangeClient = server.createArtifactoryClient(
                     preferredDeployer.provideUsername(build.getParent()), preferredDeployer.providePassword(build.getParent()),
-                    server.createProxyConfiguration(Jenkins.getInstance().proxy));
+                    preferredDeployer.provideAccessToken(build.getParent()), server.createProxyConfiguration(Jenkins.getInstance().proxy));
```
Comment (eyalbe4) [inline src/main/java/org/jfrog/hudson/pipeline/common/docker/DockerImage.java]: What is the reason for this change?

File: src/main/java/org/jfrog/hudson/pipeline/common/executors/CollectIssuesExecutor.java [hunk 3]
```diff
@@ -56,7 +56,8 @@ private ArtifactoryBuildInfoClientBuilder getBuildInfoClientBuilder(ArtifactoryS
         org.jfrog.hudson.ArtifactoryServer server = Utils.prepareArtifactoryServer(null, pipelineServer);
         CredentialsConfig preferredDeployer = CredentialManager.getPreferredDeployer(new ArtifactoryConfigurator(server), server);
         return server.createArtifactoryClientBuilder(preferredDeployer.provideUsername(build.getParent()),
-                preferredDeployer.providePassword(build.getParent()), server.createProxyConfiguration(Jenkins.getInstance().proxy), new JenkinsBuildInfoLog(listener));
+                preferredDeployer.providePassword(build.getParent()), preferredDeployer.provideAccessToken(build.getParent()),
+                server.createProxyConfiguration(Jenkins.getInstance().proxy), new JenkinsBuildInfoLog(listener));
```
Comment (eyalbe4) [inline src/main/java/org/jfrog/hudson/pipeline/common/executors/CollectIssuesExecutor.java]: The above code is pretty hard to read. I suggested making it a bit less nested or improving the code alignment.

File: src/main/java/org/jfrog/hudson/pipeline/common/executors/EditPropsExecutor.java [hunk 4]
```diff
@@ -40,6 +40,7 @@ public void execute() throws IOException, InterruptedException {
         Boolean success = ws.act(new EditPropertiesCallable(new JenkinsBuildInfoLog(listener),
             preferredDeployer.provideUsername(build.getParent()),
             preferredDeployer.providePassword(build.getParent()),
+                preferredDeployer.provideAccessToken(build.getParent()),
```
Comment (eyalbe4) [inline src/main/java/org/jfrog/hudson/pipeline/common/executors/EditPropsExecutor.java]: Can we pass a credentials object to this constructor, instead of username, password and an access token?

File: src/main/java/org/jfrog/hudson/pipeline/common/executors/GenericUploadExecutor.java [hunk 5]
```diff
@@ -46,14 +46,15 @@ public BuildInfo getBuildInfo() {
...
+        String accessToken = server.getDeployerCredentialsConfig().provideAccessToken(build.getParent());
         Credentials credentials = new Credentials(server.getDeployerCredentialsConfig().provideUsername(build.getParent()),
                 server.getDeployerCredentialsConfig().providePassword(build.getParent()));
         ProxyConfiguration proxyConfiguration = Utils.getProxyConfiguration(server);
 
         new BuildInfoAccessor(buildInfo).appendVcs(Utils.extractVcs(ws, new JenkinsBuildInfoLog(listener)));
 
         List<Artifact> deployedArtifacts = ws.act(new GenericArtifactsDeployer.FilesDeployerCallable(listener, spec,
-                server, credentials, Utils.getPropertiesMap(buildInfo, build, context), proxyConfiguration));
+                server, credentials, accessToken, Utils.getPropertiesMap(buildInfo, build, context), proxyConfiguration));
```
Comment (eyalbe4) [inline src/main/java/org/jfrog/hudson/pipeline/common/executors/GenericUploadExecutor.java]: Can we pass the accessToken as part of the credentials argument?

**Comment by eyalbe4** [comment]
@liron-shalom,
I didn't review everything, but I did submit what I have so far.
There's also a chance that you'd like to modify the code further, following the build-info changes.
Please let me know when you'd like me to review again

File: src/main/java/org/jfrog/hudson/pipeline/common/docker/DockerImage.java (hunk 1 above)
Comment (liron-shalom) [reply src/main/java/org/jfrog/hudson/pipeline/common/docker/DockerImage.java]: I had to add this parameter after I changed the constructor to get all 3 (username, password and access-token).
It's irrelevant after the current change.

File: src/main/java/org/jfrog/hudson/pipeline/common/docker/DockerImage.java (hunk 2 above)
Comment (liron-shalom) [reply src/main/java/org/jfrog/hudson/pipeline/common/docker/DockerImage.java]: I had to add this parameter after I changed the constructor to get all 3 (username, password and access-token).
It's irrelevant after the current change.

File: src/main/java/org/jfrog/hudson/CredentialsConfig.java [hunk 6]
```diff
@@ -78,37 +77,46 @@ public boolean isCredentialsProvided() {
      *
      * @return the username that should be apply in this configuration
      */
-    public String provideUsername(Item item) {
-        return isUsingCredentialsPlugin() ? PluginsUtils.credentialsLookup(credentialsId, item).getUsername() :  credentials.getUsername();
+    private String provideUsername(Item item) {
+        return isUsingCredentialsPlugin() ? PluginsUtils.usernamePasswordCredentialsLookup(credentialsId, item).getUsername() : username;
     }
     /**
      * Not like getPassword this will return the username of the current Credentials mode of the system (legacy/credentials plugin)
```
Comment (eyalbe4) [inline src/main/java/org/jfrog/hudson/CredentialsConfig.java]: ...return the username... --> ...return the password...
